### ISSUE TITLE
AirPlane Mode / Disabled Wifi detection is flaky

### ISSUE THREAD

**Issue by peetzweg** [paritytech/parity-signer#2041]
First time setting up the polkadot vault on an iphone X.

Had trouble on the first step setting to airplane mode and disbaled wifi. It wasn't detected properly, after some on/off it worked. Probably takes a while to be detected confidently. However, UX is not so good. Maybe indicate it's beeing checked if one of these two connection states change?

Could also be related to the fact, that entering airplane mode also disables wifi automatically. 

https://github.com/paritytech/parity-signer/assets/839848/63816f2d-32ce-4277-b39d-ccfd39094983

**Comment by krodak** [comment]
I'll try to find some optimisation but unfortunately we are using newest Apple's framework to asses connectivity which is `NWPathMonitor` and `usesInterfaceType` combined with `availableInterfaces`. 
Unfortunately there is no official system API that would allow us to check for "Airplane mode", so we can only validate that `availableInterfaces` is empty.

As for delay with WiFi, also there is no further system API we could use, as system provide this with delay when network connection actually cease on device itself. 

I'll have a look for some hacks, but so far can't find anything newer we could use 🙏🏻

**Comment by peetzweg** [comment]
As there is surely some delay to detecting the internet/wifi connection loss, I think the "issue" can be probably resolved with some user experience tweak. Some indicator that it's checked every 5 seconds or a continuous loading spinner. This would suggest to the user that the app is at least doing something and trying to get information about the state of connectivity.

In the current form it indicates the change would be instantly visible, as it's just a on/off state. This caused the confusion for me in the first place as the change was immediately reflected or shown as processing.

Nothing critical but felt like a bug to me personally.

**Comment by krodak** [comment]
Good idea, only implementation issue I see is that this is stream-based information, so we are not refetching anything periodically, only thing we can detect from sandbox is that application went to "background" when user brings up Control Center or go back to Home (no possibility to distinguish this...), we'll discuss it and update UI to reflect some active state.

Thanks for the feedback, this is really valuable 🙇🏻‍♂️
